Engine/ScriptAnalyzer.cs

```diff
@@ -1475,6 +1477,13 @@ public IEnumerable<DiagnosticRecord> AnalyzePath(string path, bool searchRecursi
             this.BuildScriptPathList(path, searchRecursively, scriptFilePaths);
             foreach (string scriptFilePath in scriptFilePaths)
             {
+                if (autoFix)
+                {
+                    var scriptFileContentWithAutoFixes = Fix(File.ReadAllText(scriptFilePath));
+                    // Use UTF8 when writing to avoid issues with special characters such as e.g. the copyright symbol in *.psd1 files. This could be improved to detect the encoding in order to preserve it.
+                    File.WriteAllText(scriptFilePath, scriptFileContentWithAutoFixes, Encoding.UTF8);
```

Comment:
When we modify a file, we should preserve the file encoding and line endings

Reply:
Yes, that's why I put the comment there. A quick search showed however that detecting the encoding is non-trivial. Should we do something like [this](https://stackoverflow.com/a/19283954/1810304)?

Reply:
The [`StreamReader` ](https://msdn.microsoft.com/en-us/library/system.io.streamreader.currentencoding.aspx) object provides encoding of a file. I think we should that instead of rewriting the logic to detect the encoding.